backend/spec/features/admin/returns/return_authorization_spec.rb

```diff
@@ -0,0 +1,104 @@
+    let!(:hello_user) { create(:user, email: 'hello@example.com') }
+
+    scenario 'assigns recipient to return authorization' do
+      visit spree.new_admin_order_return_authorization_path(order)
+
+      within('.return-items-table') do
+        expect(page).to have_content(Spree.t(:reimbursement_type).upcase)
+        expect(page).to have_content(Spree.t(:exchange_for).upcase)
+      end
+      expect(page).not_to have_content(Spree.t(:reimbursement_recipient).upcase)
+
+      check Spree.t(:override_recipient)
+
+      within('.return-items-table') do
+        expect(page).not_to have_content(Spree.t(:reimbursement_type).upcase)
+        expect(page).not_to have_content(Spree.t(:exchange_for).upcase)
+      end
+      expect(page).to have_content(Spree.t(:reimbursement_recipient).upcase)
+
+      open_customer_search
+
+      expect(page).to have_content('poptarts@toaster.org')
+      expect(page).to have_content('hello@example.com')
+
+      find('.customer_search .select2-input').set('poptarts@toaster.org')
+
+      expect(page).to have_content('poptarts@toaster.org')
+      expect(page).to_not have_content('hello@example.com')
+
+      find('.customer-details:first-child').click
+      find('.add-item').click
+      select 'NY Warehouse', from: 'return_authorization_stock_location_id'
+
+      expect {
+        click_button Spree.t(:create)
+      }.to change { Spree::ReturnAuthorization.count }.from(0).to(1)
+
+      return_authorization = Spree::ReturnAuthorization.last
+      expect(return_authorization.recipient).to eq(poptart_user)
+      expect(page).to have_content(Spree.t(:successfully_created, resource: 'Return Authorization'))
+    end
+
+    scenario 'without selecting another recipient' do
+      visit spree.new_admin_order_return_authorization_path(order)
+      check Spree.t(:override_recipient)
+      find('.add-item').click
+      select 'NY Warehouse', from: 'return_authorization_stock_location_id'
+
+      expect {
+        click_button Spree.t(:create)
+      }.not_to change { Spree::ReturnAuthorization.count }
+
+      expect(page).to have_content "Recipient can't be blank"
+
+      check Spree.t(:override_recipient)
+
+      open_customer_search
+      find('.customer_search .select2-input').set('hello@example.com')
+      expect(page).to have_content('hello@example.com')
+      find('.customer-details:first-child').click
+
+      find('.add-item').click
+      select 'NY Warehouse', from: 'return_authorization_stock_location_id'
+
+      expect {
+        click_button Spree.t(:create)
+      }.to change { Spree::ReturnAuthorization.count }.from(0).to(1)
+
+      return_authorization = Spree::ReturnAuthorization.last
+      expect(return_authorization.recipient).to eq(hello_user)
+      expect(page).to have_content(Spree.t(:successfully_created, resource: 'Return Authorization'))
+    end
+  end
+
+  def open_customer_search
+    execute_script('$("#customer_search").select2("open")')
```

Comment:
Yeah we had trouble opening the customer search box. We really didn't want to use `execute_script` as well. We'll try your suggestion instead. :)